centreon/src/Core/HostCategory/Infrastructure/Repository/DbReadHostCategoryRepository.php

```diff
@@ -0,0 +1,203 @@
+        ]);
+        $sqlTranslator?->addNormalizer('is_activated', new BoolToEnumNormalizer());
+        $sqlTranslator?->translateForConcatenator($concatenator);
+
+        $statement = $this->db->prepare($this->translateDbName($concatenator->__toString()));
+
+        $sqlTranslator?->bindSearchValues($statement);
+        foreach ($concatenator->retrieveBindValues() as $param => [$value, $type]) {
+            $statement->bindValue($param, $value, $type);
+        }
+        $statement->execute();
+
+        $sqlTranslator?->calculateNumberOfRows($this->db);
+
+        $hostCategories = [];
+        while (is_array($result = $statement->fetch(\PDO::FETCH_ASSOC))) {
+            /** @var array{hc_id:int,hc_name:string,hc_alias:string,hc_activate:'0'|'1',hc_comment:string|null} $result */
+            $hostCategories[] = $this->createHostCategoryFromArray($result);
+        }
+
+        return $hostCategories;
+    }
+
+    /**
+     * @param array{hc_id:int,hc_name:string,hc_alias:string,hc_activate:'0'|'1',hc_comment:string|null} $result
+     * @return HostCategory
+     */
+    private function createHostCategoryFromArray(array $result): HostCategory
+    {
+        $hostCategory = new HostCategory(
+            $result['hc_id'],
+            $result['hc_name'],
+            $result['hc_alias']
+        );
+        $hostCategory->setActivated($result['hc_activate'] === self::HC_IS_ACTIVE ? true : false);
+        $hostCategory->setComment($result['hc_comment']);
+
+        return $hostCategory;
+    }
+
+    /**
+     * Determine if host cateogries are filtered for given access group ids
+     * true: accessible host categories are filtered
+     * false: accessible host categories are not filtered
+     *
+     * @param int[] $accessGroupIds
+     * @phpstan-param non-empty-array<int> $accessGroupIds
+     * @return bool
+     */
+    private function hasRestrictedAccessToHostCategories(array $accessGroupIds): bool
+    {
+        $concatenator = new SqlConcatenator();
+
+        $concatenator->defineSelect(
+            'SELECT 1
+            FROM `:db`.acl_resources_hc_relations arhr
+            INNER JOIN `:db`.acl_resources res
+                ON arhr.acl_res_id = res.acl_res_id
+            INNER JOIN `:db`.acl_res_group_relations argr
+                ON res.acl_res_id = argr.acl_res_id
+            INNER JOIN `:db`.acl_groups ag
+                ON argr.acl_group_id = ag.acl_group_id'
+        );
+
+        $concatenator->storeBindValueMultiple(':access_group_ids', $accessGroupIds, \PDO::PARAM_INT)
+            ->appendWhere('ag.acl_group_id IN (:access_group_ids)');
+
+        $statement = $this->db->prepare($this->translateDbName($concatenator->__toString()));
+
+        foreach ($concatenator->retrieveBindValues() as $param => [$value, $type]) {
+            $statement->bindValue($param, $value, $type);
+        }
+        $statement->execute();
+
+        return (bool) ($statement->fetchColumn());
```

Comment:
Useless parentheses.
```suggestion
        return (bool) $statement->fetchColumn();
```